src/client/pkg/config/config.proto

```diff
@@ -24,6 +24,13 @@ message Config {
 // DO NOT change or remove field numbers from this proto, as if you do, v1 user
 // configs will become unparseable.
 message ConfigV1 {
```

Comment:
The rationale is that it makes replacing "configV1" much less awkward (keep in mind that the name "ConfigV1" won't appear in peoples' configs. Just the field name "v1")

Basically, suppose at some point in the future we want to change the structure of configs (what I had in mind was changing it so that a pachcfg had all of a user's past cluster, and then a current-cluster that points at one of them, similar to kubecfg). We can't remove the existing fields from the proto, because that would cause peoples' existing configs to fail to parse (they'd have a bunch of unrecognized field names). So we'd either have a bunch of fields in the top-level message (in this example, `pachd-address`, `repeated cluster` and `current-cluster` where either the first is set or the last two are, and if either of the last two is set they must both be set)

In the long run, we'd have a lot of fields that don't make sense because they're archaic. By having explicit versions, we'd have several different versions of the config, all of which are internally consistent, and exactly one of which is set.

(that's the rationale, at least)